component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/token/binder/DPoPBasedTokenBinder.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.dpop.token.binder;
+
+import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants.GrantTypes;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AccessTokenReqDTO;
+import org.wso2.carbon.identity.oauth2.token.bindings.impl.AbstractTokenBinder;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This class provides the DPoP based token binder implementation.
+ */
+public class DPoPBasedTokenBinder extends AbstractTokenBinder {
+
+    private static final String BINDING_TYPE = "DPoP";
+    private static Optional<String> tokenBindingValue;
+    private final List<String> supportedGrantTypes = Arrays.asList(GrantTypes.AUTHORIZATION_CODE, GrantTypes.PASSWORD,
```

Comment:
Is it supported in all these grant types?

Reply:
As per the  [spec](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-dpop-02#section-5)  The DPoP token requests are  applicable for all access token requests  regardless of the grant type.

Reply:
Did you test this with all grant types?

Reply:
Yes